app/src/main/java/com/ashlett/memory/GamePresenter.kt

```diff
@@ -0,0 +1,29 @@
+package com.ashlett.memory
+
+interface Presenter<in T> {
```

Comment:
I would move this interface into its own class, then you could reuse it in other views.